generator/main.go

```diff
@@ -0,0 +1,227 @@
+	// each turns into a command line option,
+	// and has options for output forms.
+	allGenerators = map[string]genall.Generator{
+		"overrides":  overrides.Generator{},
+		"interfaces": interfaces.Generator{},
+		"crds":       crds.Generator{},
+		"deepcopy":   deepcopy.Generator{},
+		"schemas":    schemas.Generator{},
+	}
+
+	// allOutputRules defines the list of all known output rules, giving
+	// them names for use on the command line.
+	// Each output rule turns into two command line options:
+	// - output:<generator>:<form> (per-generator output)
+	// - output:<form> (default output)
+	allOutputRules = map[string]genall.OutputRule{
+		"dir":       genall.OutputToDirectory(""),
+		"none":      genall.OutputToNothing,
+		"stdout":    genall.OutputToStdout,
+		"artifacts": genall.OutputArtifacts{},
+	}
+
+	// optionsRegistry contains all the marker definitions used to process command line options
+	optionsRegistry = &markers.Registry{}
+)
+
+func init() {
+	for genName, gen := range allGenerators {
+		// make the generator options marker itself
+		defn := markers.Must(markers.MakeDefinition(genName, markers.DescribesPackage, gen))
+		if err := optionsRegistry.Register(defn); err != nil {
+			panic(err)
+		}
+		if helpGiver, hasHelp := gen.(genall.HasHelp); hasHelp {
+			if help := helpGiver.Help(); help != nil {
+				optionsRegistry.AddHelp(defn, help)
+			}
+		}
+
+		// make per-generation output rule markers
+		for ruleName, rule := range allOutputRules {
+			ruleMarker := markers.Must(markers.MakeDefinition(fmt.Sprintf("output:%s:%s", genName, ruleName), markers.DescribesPackage, rule))
+			if err := optionsRegistry.Register(ruleMarker); err != nil {
+				panic(err)
+			}
+			if helpGiver, hasHelp := rule.(genall.HasHelp); hasHelp {
+				if help := helpGiver.Help(); help != nil {
+					optionsRegistry.AddHelp(ruleMarker, help)
+				}
+			}
+		}
+	}
+
+	// make "default output" output rule markers
+	for ruleName, rule := range allOutputRules {
+		ruleMarker := markers.Must(markers.MakeDefinition("output:"+ruleName, markers.DescribesPackage, rule))
+		if err := optionsRegistry.Register(ruleMarker); err != nil {
+			panic(err)
+		}
+		if helpGiver, hasHelp := rule.(genall.HasHelp); hasHelp {
+			if help := helpGiver.Help(); help != nil {
+				optionsRegistry.AddHelp(ruleMarker, help)
+			}
+		}
+	}
+
+	// add in the common options markers
+	if err := genall.RegisterOptionsMarkers(optionsRegistry); err != nil {
+		panic(err)
+	}
+}
+
+// noUsageError suppresses usage printing when it occurs
+// (since cobra doesn't provide a good way to avoid printing
+// out usage in only certain situations).
+type noUsageError struct{ error }
+
+func main() {
+	helpLevel := 0
+	whichLevel := 0
+	showVersion := false
+
+	cmd := &cobra.Command{
+		Use:   "generate",
+		Short: "Generate Kubernetes API extension resources and code.",
+		Long:  "Generate Kubernetes API extension resources and code.",
+		Example: `	# Provide the right examples ...`,
```

Comment:
Any plan to provide the right example?

Reply:
I just missed it. Thanks for pointing this out.
This was fixed in commit https://github.com/devfile/api/pull/133/commits/66c0dd08da2c596e5c77995eefd59a9579975fc0